game/src/graphic/textures/TextureHandler.java

```diff
@@ -2,30 +2,67 @@
 
 import com.badlogic.gdx.Gdx;
 import com.badlogic.gdx.files.FileHandle;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
 import java.util.Arrays;
+import java.util.Comparator;
 import java.util.LinkedHashMap;
 import java.util.LinkedHashSet;
 import java.util.List;
 import java.util.Map;
 import java.util.Set;
+import java.util.function.Predicate;
 import java.util.regex.Pattern;
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
 
 /**
  * A texture handler class for managing textures paths for further use.
  *
- * <p>Singleton (possibly thread-safe).
+ * <p>This class is designed as Singleton, because all asset paths should be read only once at the
+ * beginning of the application.
  */
 public class TextureHandler {
+    public static final String PLACEHOLDER_FILENAME = ".resource_root";
+
     private static final TextureHandler INSTANCE = new TextureHandler();
 
     private final Map<String, Set<FileHandle>> pathMap = new LinkedHashMap<>();
 
     private TextureHandler() {
-        addAllAssets(Gdx.files.internal(Gdx.files.getLocalStoragePath()));
+        try {
+            addAllAssets(getResourceRoot());
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    /**
+     * Searches for the resource root that has the longest path name string. Internal helper method.
+     *
+     * @return the resource root with the longest path name string.
+     */
+    private FileHandle getResourceRoot() throws IOException {
+        Predicate<Path> isPlaceHolder =
+                p -> PLACEHOLDER_FILENAME.equals(p.getFileName().toString());
+        // We assume that the build directory structure looks like this: Dungeon/<build
+        // dir>/resources/main/<...>. So we need to dive at least into the directories with a depth
+        // of 4.
```

Comment:
@tgrothe Kannst Du das bitte in den Javadoc der Methode verschieben?

Reply:
hab ich